tests/on_target/tests/test_memfault.py

```diff
@@ -0,0 +1,84 @@
+import pytest
+import os
+import requests
+import datetime
+import time
+from utils.flash_tools import flash_device, reset_device
+import sys
+sys.path.append(os.getcwd())
+from utils.logger import get_logger
+
+MEMFAULT_ORG_TOKEN = os.getenv('MEMFAULT_ORGANIZATION_TOKEN')
+MEMFAULT_ORG = os.getenv('MEMFAULT_ORGANIZATION_SLUG')
+MEMFAULT_PROJ = os.getenv('MEMFAULT_PROJECT_SLUG')
+IMEI = os.getenv('IMEI')
+MEMFAULT_TIMEOUT = 5 * 60
+
+logger = get_logger()
+
+url = "https://api.memfault.com/api/v0"
+auth = ("", MEMFAULT_ORG_TOKEN)
+
+def get_event_log():
+    response = requests.get(f"{url}/{MEMFAULT_ORG}/{MEMFAULT_PROJ}/event-log", auth=auth)
+    response.raise_for_status()
+    return response.json()
+
+def get_latest_events(event_type, device_id):
+    data = get_event_log()
+    latest_events = [
+        x
+        for x in data["data"]
+        if x["device_serial"] == str(device_id) and x["type"] == event_type
+    ]
+    return latest_events
+
+def timestamp(event):
+    return datetime.datetime.strptime(
+        event["captured_date"], "%Y-%m-%dT%H:%M:%S.%f%z"
+    )
+
+def wait_for_heartbeat(timestamp_old_reboot_evt):
+    new_heartbeat_found = False
+    start = time.time()
+    while time.time() - start < MEMFAULT_TIMEOUT:
+        logger.debug("Looking for latest heartbeat events")
+        time.sleep(5)
+        new_heartbeat_events = get_latest_events("HEARTBEAT", IMEI)
+        if not new_heartbeat_events:
+            continue
+        if not new_heartbeat_events[0]["summary"]["event_info"]["metrics"]["MemfaultSdkMetric_UnexpectedRebootCount"] == 1:
```

Comment:
Perhaps this should be wrapped in a `try/except (KeyError, ValueError, ...): continue` statement?

Can this `UnexpectedRebootCount` increase past 1 when the test is run multiple times?